case-study.Rmd

```diff
@@ -0,0 +1,467 @@
+```
+
+Combining the volcano plot and the activity heatmap, Node35pos is the most
+active signature in anr mutant, followed by Node233pos, Node140pos.
+On the other side, Node38neg, Node31pos, Node269pos, Node205neg are most active
+in wildtype.
+
+## Overlapping signature removal
+To reduce the number of signatures to look at, we can check whether these active
+signatures overlap with each other.
+
+`plot_signature_overlap` creates a heatmap of odds ratios. The odds ratio
+represents the odds that two signatures share a specific number of genes.
+
+```{r}
+signature_similarity <- plot_signature_overlap(selected_signatures = active_sigs,
+                                               model = model)
+```
+
+Next we calculate the marginal activities of similar signatures. Marginal
+activity is defined as the activity of signature A after removing genes that it
+shares with signature B.
+
+```{r}
+marginal_activity <- calculate_marginal_activity(input_data = data_normed,
+                                                 selected_signatures = active_sigs,
+                                                 model = model)
+```
+
+Again, we build a limma model to test whether these marginal activities
+are still strongly different between two conditions.
+
+```{r}
+marginal_limma <- build_limma(input_data = marginal_activity,
+                              phenotypes = data_pheno, control_pheno = "wt")
+```
+
+Let's visualize the marginal activities in a matrix heatmap.  The value in this
+matrix represents the -log10 transformed adjusted p-value in
+the activation test when the effect of the column signature is removed from
+the row signature. Values in the diagonal of the heatmap are the activation
+significance of signatures themselves. Activation significance below the
+significance cutoff is marked by a cross sign.
+
+```{r}
+plot_marginal_activation(marginal_limma_result = marginal_limma,
+                         signature_order = colnames(signature_similarity),
+                         sig_cutoff = 0.05)
+
+```
+
+Based on this plot, we can see that Node119pos, Node214neg, and Node299pos are
+completely masked by Node191pos, because after removing Node191pos from them,
+they become non-significant. Node130pos and Node250neg are masked by
+Node31pos. Node285neg is masked by Node205neg.
+Node154pos is masked by Node57neg. Node63neg, Node39neg, Node228pos, Node158neg,
+Node140pos, Node269neg and Node31neg are masked by Node35pos. Node185neg and
+Node275pos are masked by Node67pos. Node278pos is masked by Node9pos.
+We can see that Node67pos, Node35pos, and Node233pos each has unique
+genes that make them still significant even after removing the effect
+of another signature. We can safely remove a signature being masked by another
+signature as long as we keep the second signature.
+
+```{r}
+unique_active_sigs <- remove_redundant_signatures(marginal_limma,
+                                                  sig_cutoff = 0.05)
+unique_active_sigs
```

Comment:
`unique_active_sigs` is a character vector, it only lists the name of signatures that are active and unique.